networkAware update configs

jpedro1992 opened the pull request:
#### What type of PR is this?

Discuss and solve issue #461 about the config and deployment files of the networkAware plugins. 

#### What this PR does / why we need it:

The deployment files included in the manifest folder have been updated. Also, the networkAware plugins have been made visible in the main README file.  

#### Which issue(s) this PR fixes:
Fixes #461

ttsokov commented on manifests/networktopology/deploy.yaml:
```diff
@@ -0,0 +1,50 @@
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      serviceAccountName: network-aware-scheduler
```
The following rules have to be added into the ClusterRole, which is bound to the ServiceAccount with name `network-aware-scheduler`:
```
- apiGroups: ["appgroup.diktyo.k8s.io"]
  resources: ["appgroups"]
  verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
- apiGroups: ["networktopology.diktyo.k8s.io"]
  resources: ["networktopologies"]
  verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
```
Probably RBAC yaml file containing the ServiceAccount, ClusterRole, ClusterRoleBinding, Role and RoleBinding can be added inside `manifests/networktopology/` directory, similar to this one - https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/manifests/noderesourcetopology/cluster-role.yaml?

ttsokov commented on manifests/appgroup/deploy.yaml:
```diff
@@ -0,0 +1,50 @@
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      serviceAccountName: network-aware-scheduler
```
The following rules have to be added into the ClusterRole, which is bound to the ServiceAccount with name `network-aware-scheduler`:
```
- apiGroups: ["appgroup.diktyo.k8s.io"]
  resources: ["appgroups"]
  verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
- apiGroups: ["networktopology.diktyo.k8s.io"]
  resources: ["networktopologies"]
  verbs: ["get", "list", "watch", "create", "delete", "update", "patch"]
```
Probably RBAC yaml file containing the ServiceAccount, ClusterRole, ClusterRoleBinding, Role and RoleBinding can be added inside `manifests/appgroup/` directory, similar to this one - https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/manifests/noderesourcetopology/cluster-role.yaml?

jpedro1992 replied:
Thank you for suggestion @ttsokov! I will update the PR accordingly.

Huang-Wei commented:
/ok-to-test

Huang-Wei reviewed (commented):
Thanks @jpedro1992 . Could you also add the RBAC bits to manifests/install/charts/as-a-second-scheduler/templates/rbac.yaml, and then modify manifests/install/charts/as-a-second-scheduler/values.yaml locally (don't need to check in) to see if Helm install and your plugin function well?

jpedro1992 commented:
Thanks @Huang-Wei for the review. I will update the PR.

jpedro1992 commented:
Hi @Huang-Wei and @ttsokov, 

Thank you for your review! I added a couple of deployment files (appgroup-controller, networktopology-controller, etc) and a few comments to a few files in order to properly deploy the scheduler with network-aware plugins. 

A few notes below: 

- Building the image locally and running it on K8s v1.23.X does not work. I face the following error: 

`"E0130 21:35:32.938159       1 reflector.go:138] [k8s.io/client-go/informers/factory.go:134](http://k8s.io/client-go/informers/factory.go:134): Failed to watch *v1.CSIStorageCapacity: failed to list *v1.CSIStorageCapacity: the server could not find the requested resource".` 

All pods deployed with the scheduler are stuck in a pending state. 

- Building the image locally or using the registry.k8s.io/scheduler-plugins/kube-scheduler:v0.24.9 and running it on K8s v1.24.4 works. I added a comment mentioning that network aware plugins should be used with scheduler plugins v.0.24.9

Huang-Wei commented:
> Building the image locally and running it on K8s v1.23.X does not work

That's not surprising given you're running a high-version scheduler (for master it's 1.25) against to a low-version API Server (1.23).

Huang-Wei commented:
/lgtm
/approve

/release-note-none

Huang-Wei commented:
/hold

@jpedro1992 could you take a look at the CI failure?

jpedro1992 commented:
Hi @Huang-Wei, I rebased the PR and everything seems fine now. For some reason, it was not assuming the previous commit that had solved the problem with the integration test.

Huang-Wei commented:
/lgtm
/unhold
